target_bigquery.py

```diff
@@ -355,7 +352,23 @@ def write_rows_to_bigquery(streams, emit_state_after_write=False):
                                 fixed_rows[number_of_rows // 2 :],
                                 row_ids=ids[number_of_rows // 2 :],
                             )
-                        elif getattr(e, "errors", [{}])[0].get("reason") in RETRYABLE_ERROR_CODES:
+
+                        if (
+                            "payload size exceeds the limit" in error_string
+                            or "too many rows present" in error_string
+                        ):
+                            insert_in_halves()
+                        # While this is similar to the above, we want to avoid doing the json.dumps
+                        # if possible, so will only do it if we didn't get the expected errors
+                        elif len(json.dumps(fixed_rows)) > 10000000:
```

Comment:
Didn't we have a problem with this where we tried 5k rows but still hit an error as the rows were very wide?

Reply:
This is the condition for that case, it dumps the JSON and checks the length so we'd see if it's likely to be above the 10MB threshold.